File: app/src/main/java/org/jellyfin/androidtv/data/querying/StdItemQuery.java
package org.jellyfin.androidtv.data.querying;

import org.jellyfin.androidtv.auth.UserRepository;
import org.jellyfin.apiclient.model.querying.ItemFields;
import org.jellyfin.apiclient.model.querying.ItemQuery;
import org.koin.java.KoinJavaComponent;

public class StdItemQuery extends ItemQuery {
    public StdItemQuery(ItemFields[] fields) {
        if (fields == null) {
            fields = new ItemFields[]{
                    ItemFields.PrimaryImageAspectRatio,
                    ItemFields.Overview,
                    ItemFields.ItemCounts,
                    ItemFields.DisplayPreferencesId,
                    ItemFields.ChildCount
            };
        }
        setUserId(KoinJavaComponent.<UserRepository>get(UserRepository.class).getCurrentUser().getValue().getId().toString());
        setFields(fields);
    }

    public StdItemQuery() {
        this(null);
    }
}

File: app/src/main/java/org/jellyfin/androidtv/data/querying/StdItemQuery.kt
package org.jellyfin.androidtv.data.querying

import org.jellyfin.androidtv.auth.repository.UserRepository
import org.jellyfin.apiclient.model.querying.ItemFields
import org.jellyfin.apiclient.model.querying.ItemQuery
import org.koin.core.component.KoinComponent
import org.koin.core.component.get

class StdItemQuery @JvmOverloads constructor(
	fields: Array<ItemFields> = defaultFields
) : ItemQuery(), KoinComponent {
	companion object {
		val defaultFields = arrayOf(
			ItemFields.PrimaryImageAspectRatio,
			ItemFields.Overview,
			ItemFields.ItemCounts,
			ItemFields.DisplayPreferencesId,
			ItemFields.ChildCount
		)
	}

	init {
		userId = get<UserRepository>().currentUser.value!!.id.toString()
		setFields(fields)
	}
}
